PlayerMovement.cs

```diff
@@ -1,180 +1,166 @@
 // Some stupid rigidbody based movement by Dani
 
-using System;
 using UnityEngine;
 
 public class PlayerMovement : MonoBehaviour {
 
+    Rigidbody rigidBody;
+
     //Assingables
     public Transform playerCam;
     public Transform orientation;
-    
-    //Other
-    private Rigidbody rb;
 
     //Rotation and look
-    private float xRotation;
-    private float sensitivity = 50f;
-    private float sensMultiplier = 1f;
-    
+    float xRotation;
+    const float sensitivity = 50f;
+
     //Movement
     public float moveSpeed = 4500;
     public float maxSpeed = 20;
-    public bool grounded;
-    public LayerMask whatIsGround;
-    
-    public float counterMovement = 0.175f;
-    private float threshold = 0.01f;
+    bool grounded;
+
+    const float threshold = 0.01f;
+    public float friction = 0.175f;
     public float maxSlopeAngle = 35f;
 
     //Crouch & Slide
-    private Vector3 crouchScale = new Vector3(1, 0.5f, 1);
-    private Vector3 playerScale;
+    Vector3 crouchScale = new Vector3(1, 0.5f, 1);
+    Vector3 playerScale;
     public float slideForce = 400;
-    public float slideCounterMovement = 0.2f;
+    public float slideFriction = 0.2f;
 
     //Jumping
-    private bool readyToJump = true;
-    private float jumpCooldown = 0.25f;
     public float jumpForce = 550f;
-    
+
     //Input
-    float x, y;
-    bool jumping, sprinting, crouching;
-    
+    Vector2 inputDirection = new Vector2();
```

Comment:
Keeping input in a `Vector2` means we can normalize the input to avoid diagonal movement being 2x desired speed. Jumping wasn't necessary after this change, and sprinting was unused.